src/WebApiContrib.Core/MvcOptionsExtensions.cs

```diff
@@ -24,5 +29,44 @@ public static void UseGlobalRoutePrefix(this MvcOptions opts, IRouteTemplateProv
         {
             opts.Conventions.Insert(0, new FromBodyApplicationModelConvention(controllerPredicate, actionPredicate, parameterPredicate));
         }
+
+        /// <summary>
+        /// Registers a Validation action filter as application model convention, which applies this filter to POST, PUT, and DELETE actions.
+        /// </summary>
+        /// <param name="controllerPredicate">A controller predicate for further configuration.</param>
+        /// <param name="actionPredicate">An action predicate for further configuration.</param>
+        public static void UseValidationFilter(this MvcOptions opts, Func<ControllerModel, bool> controllerPredicate = null, Func<ActionModel, bool> actionPredicate = null)
+    => UseValidationFilter(opts, new ValidationAttribute(), controllerPredicate, actionPredicate);
+
+        /// <summary>
+        /// Registers a Validation action filter as application model convention, which applies this filter to POST, PUT, and DELETE actions.
+        /// </summary>
+        /// <param name="filter">A filter of type ValidationAttribute.</param>
+        /// <param name="controllerPredicate">A controller predicate for further configuration.</param>
+        /// <param name="actionPredicate">An action predicate for further configuration.</param>
+        public static void UseValidationFilter(this MvcOptions opts, ValidationAttribute filter, Func<ControllerModel, bool> controllerPredicate = null, Func<ActionModel, bool> actionPredicate = null)
+        {
+            // Exclude GET and HEAD actions
+            Func<SelectorModel, bool> selectorPredicate = (s =>
+            s.ActionConstraints.OfType<HttpMethodActionConstraint>().Any(c => !c.HttpMethods.Contains(HttpMethod.Get.Method)
+     || !c.HttpMethods.Contains(HttpMethod.Head.Method)));
+
+            if (filter.GetType() != typeof(ValidationAttribute)) throw new ArgumentException($"Do not use this method to add other action filters than Validation action filters. Use the '{nameof(AddGlobalFilter)}' extension method for other action filter registrations.");
```

Comment:
this type check is unnecessary. Since the parameter is `ValidationAttribute`, you can't pass anything else but `ValidationAttribute` and all of classes derived from it (which is OK)